zipline/data/minute_bars.py

```diff
@@ -229,41 +229,39 @@ def read(cls, rootdir):
                 # if version does not match.
                 version = 0
 
-            first_trading_day = pd.Timestamp(
-                raw_data['first_trading_day'], tz='UTC')
-            market_opens = pd.to_datetime(raw_data['market_opens'],
-                                          unit='m',
-                                          utc=True)
-            market_closes = pd.to_datetime(raw_data['market_closes'],
-                                           unit='m',
-                                           utc=True)
+            start_session = pd.Timestamp(raw_data['start_session'], tz='UTC')
+            end_session = pd.Timestamp(raw_data['end_session'], tz='UTC')
+
             ohlc_ratio = raw_data['ohlc_ratio']
 
             if version == 0:
                 # version 0 always assumed US equities.
                 minutes_per_day = US_EQUITIES_MINUTES_PER_DAY
+                # version 0 always assumed NYSE.
+                calendar = get_calendar('NYSE')
             else:
                 minutes_per_day = raw_data['minutes_per_day']
+                calendar = get_calendar(raw_data['calendar_name'])
```

Comment:
This may need a check for `version < 2` (and an increment of the version.), so that Zipline can be updated in places where the file format is read, ahead of the new key being generated on the data set for the first time.